docs/src/examples/Hierarchical-abstraction.jl

```diff
@@ -0,0 +1,194 @@
+    x_prev = concrete_system.f_backward(x, u)
+    xpos_cell = DO.get_pos_by_coord(grid, x_prev)
+    n = 2
+    for i in (-n):n
+        for j in (-n):n
+            x_n = (xpos_cell[1] + i, xpos_cell[2] + j)
+            x_n = DO.set_in_period_pos(abstract_system.Xdom, x_n)
+            if x_n in abstract_system.Xdom
+                cell = SY.get_state_by_xpos(abstract_system, x_n)[1]
+                if !(cell in potential)
+                    push!(potential, cell)
+                end
+            end
+        end
+    end
+    return potential
+end
+
+function compute_reachable_set(rect::UT.HyperRectangle, concrete_system, Udom)
+    r = (rect.ub - rect.lb) / 2.0 + concrete_system.measnoise
+    Fr = r
+    x = UT.get_center(rect)
+    n = UT.get_dims(rect)
+    lb = fill(Inf, n)
+    ub = fill(-Inf, n)
+    for upos in DO.enum_pos(Udom)
+        u = DO.get_coord_by_pos(Udom.grid, upos)
+        Fx = concrete_system.f_eval(x, u)
+        lb = min.(lb, Fx .- Fr)
+        ub = max.(ub, Fx .+ Fr)
+    end
+    lb = SVector{n}(lb)
+    ub = SVector{n}(ub)
+    return UT.HyperRectangle(lb, ub)
+end
+
+minimum_transition_cost(symmodel, contsys, source, target) = 1.0
+
+concrete_problem = SimpleProblem.problem(;
+    rectX = UT.HyperRectangle(SVector(0.0, 0.0), SVector(60.0, 60.0)),
+    obstacles = [UT.HyperRectangle(SVector(22.0, 21.0), SVector(25.0, 32.0))],
+    periodic = Int[],
+    periods = [30.0, 30.0],
+    T0 = [0.0, 0.0],
+    rectU = UT.HyperRectangle(SVector(-2.0, -2.0), SVector(2.0, 2.0)),
+    Uobstacles = [UT.HyperRectangle(SVector(-0.5, -0.5), SVector(0.5, 0.5))],
+    _I_ = UT.HyperRectangle(SVector(6.5, 6.5), SVector(7.5, 7.5)),
+    _T_ = UT.HyperRectangle(SVector(44.0, 43.0), SVector(49.0, 48.0)),
+    state_cost = UT.ZeroFunction(),
+    transition_cost = UT.ConstantControlFunction(1.0),
+    tstep = 0.8,
+    measnoise = SVector(0.0, 0.0),
+)
+
+concrete_system = concrete_problem.system;
+
+# Local optimizer parameters
+hx_local = [0.5, 0.5]
+hx_heuristic = [1.0, 1.0]
+u0 = SVector(0.0, 0.0)
+hu = SVector(0.5, 0.5)
+Ugrid = DO.GridFree(u0, hu)
+
+local_optimizer = MOI.instantiate(AB.LazyAbstraction.Optimizer)
+
+AB.LazyAbstraction.set_optimizer_parameters!(
+    local_optimizer,
+    100,
+    pre_image,
+    post_image,
+    compute_reachable_set,
+    minimum_transition_cost,
+    hx_local,
+    hx_heuristic;
+    γ = 10.0,
+)
+
+# Global optimizer parameters
+hx_global = [10.0, 10.0] #[15.0, 15.0]
+u0 = SVector(0.0, 0.0)
+hu = SVector(0.5, 0.5)
+Ugrid = DO.GridFree(u0, hu)
+max_iter = 6 # 9
+max_time = 1000
+
+optimizer = MOI.instantiate(AB.HierarchicalAbstraction.Optimizer)
+
+AB.HierarchicalAbstraction.set_optimizer!(
+    optimizer,
+    concrete_problem,
+    hx_global,
+    Ugrid,
+    compute_reachable_set,
+    minimum_transition_cost,
+    local_optimizer,
+    max_iter,
+    max_time;
+    option = true,
+)
+
+using Suppressor
+@suppress begin # this is a workaround to supress the undesired output of SDPA
```

Comment:
Didn't know about the trick, it's nice

Reply:
yes, this eliminates all your own print and those of the functions you use that you might have during the execution.